### PR TITLE
Add Teleport reward type

### PR DESCRIPTION
**Pull request by lijinhong11**

### CODE REVIEW

File: src/main/java/org/black_ixx/bossshop/core/rewards/BSRewardTypeTeleport.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        return InputReader.readString(o, true);
+    }
+
+    @Override
+    public boolean validityCheck(String item_name, Object o) {
+        String str = o.toString();
+        if (str != null&&!str.isBlank()){
+            return true;
+        }
+        ClassManager.manager.getBugFinder().severe("Was not able to create ShopItem " + item_name + "! The reward object needs no to be a empty string.");
```
Comment (Travja) [inline src/main/java/org/black_ixx/bossshop/core/rewards/BSRewardTypeTeleport.java:24]: Just the grammar police in me, but this should probably say "The reward object cannot be an empty string"

File: src/main/java/org/black_ixx/bossshop/core/rewards/BSRewardTypeTeleport.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+            else ClassManager.manager.getMessageHandler().sendMessage("Main.WorldNotExisting",p);
+        } else if (strings.length==6) {
+            World w = Bukkit.getWorld(strings[0]);
+            if (w != null) {
+                p.teleport(new Location(w, Double.parseDouble(strings[1]), Double.parseDouble(strings[2]), Double.parseDouble(strings[3]),
+                        Float.parseFloat(strings[4]), Float.parseFloat(strings[5])));
+            }
+            else ClassManager.manager.getMessageHandler().sendMessage("Main.WorldNotExisting",p);
+        } else {
+            ClassManager.manager.getBugFinder().warn("Unable to teleport "+p.getName()+" to "+reward+". Maybe the string use wrong format.");
```
Comment (Travja) [inline src/main/java/org/black_ixx/bossshop/core/rewards/BSRewardTypeTeleport.java:57]: Again, just grammar. "The string's format may be incorrect." probably works better

**Review by Travja — CHANGES_REQUESTED** [review]
Just grammatical issues. But other than that, it looks good!

File: src/main/java/org/black_ixx/bossshop/core/rewards/BSRewardTypeTeleport.java (hunk 2 above)
Comment (lijinhong11) [reply src/main/java/org/black_ixx/bossshop/core/rewards/BSRewardTypeTeleport.java:57]: sorry my english bad

File: src/main/java/org/black_ixx/bossshop/core/rewards/BSRewardTypeTeleport.java (hunk 1 above)
Comment (lijinhong11) [reply src/main/java/org/black_ixx/bossshop/core/rewards/BSRewardTypeTeleport.java:24]: sorry my english bad

**Comment by lijinhong11** [comment]
idk how to fix #18

**Review by Travja — APPROVED** [review]

**Pull request closed without merging**